src/measure_theory/content.lean

```diff
@@ -0,0 +1,210 @@
+bsupr_le $ λ K' hK', h _ _ (subset.trans hK' h2)
+
+lemma inner_content_of_is_compact {μ : compacts G → ennreal}
+  (h : ∀ (K₁ K₂ : compacts G), K₁.1 ⊆ K₂.1 → μ K₁ ≤ μ K₂) {K : set G} (h1K : is_compact K)
+  (h2K : is_open K) : inner_content μ ⟨K, h2K⟩ = μ ⟨K, h1K⟩ :=
+le_antisymm (bsupr_le $ λ K' hK', h _ ⟨K, h1K⟩ hK')
+            (le_inner_content _ _ subset.rfl)
+
+lemma inner_content_empty {μ : compacts G → ennreal} (h : μ ⊥ = 0) :
+  inner_content μ ∅ = 0 :=
+begin
+  refine le_antisymm _ (zero_le _), rw ←h,
+  refine bsupr_le (λ K hK, _),
+  have : K = ⊥, { ext1, rw [subset_empty_iff.mp hK, compacts.bot_val] }, rw this, refl'
+end
+
+/-- This is "unbundled", because that it required for the API of `induced_outer_measure`. -/
+lemma inner_content_mono {μ : compacts G → ennreal} ⦃U V : set G⦄ (hU : is_open U) (hV : is_open V)
+  (h2 : U ⊆ V) : inner_content μ ⟨U, hU⟩ ≤ inner_content μ ⟨V, hV⟩ :=
+supr_le_supr $ λ K, supr_le_supr_const $ λ hK, subset.trans hK h2
+
+lemma inner_content_exists_compact {μ : compacts G → ennreal} {U : opens G}
+  (hU : inner_content μ U < ⊤) {ε : nnreal} (hε : 0 < ε) :
+  ∃ K : compacts G, K.1 ⊆ U ∧ inner_content μ U ≤ μ K + ε :=
+begin
+  have h'ε := ennreal.zero_lt_coe_iff.2 hε,
+  cases le_or_lt (inner_content μ U) ε,
+  { exact ⟨⊥, empty_subset _, le_trans h (le_add_of_nonneg_left (zero_le _))⟩ },
+  have := ennreal.sub_lt_sub_self (ne_of_lt hU) (ne_of_gt $ lt_trans h'ε h) h'ε,
+  conv at this {to_rhs, rw inner_content }, simp only [lt_supr_iff] at this,
+  rcases this with ⟨U, h1U, h2U⟩, refine ⟨U, h1U, _⟩,
+  rw [← ennreal.sub_le_iff_le_add], exact le_of_lt h2U
+end
+
+lemma inner_content_Sup_nat [t2_space G] {μ : compacts G → ennreal}
+  (h1 : μ ⊥ = 0)
+  (h2 : ∀ (K₁ K₂ : compacts G), μ (K₁ ⊔ K₂) ≤ μ K₁ + μ K₂) (U : ℕ → opens G) :
+  inner_content μ (⨆ (i : ℕ), U i) ≤ ∑' (i : ℕ), inner_content μ (U i) :=
+begin
+  have h3 : ∀ (t : finset ℕ) (K : ℕ → compacts G), μ (t.sup K) ≤ t.sum (λ i, μ (K i)),
+  { intros t K, refine finset.induction_on t _ _,
+    { simp only [h1, le_zero_iff_eq, finset.sum_empty, finset.sup_empty] },
+    { intros n s hn ih, rw [finset.sup_insert, finset.sum_insert hn],
+      exact le_trans (h2 _ _) (add_le_add_left ih _) }},
+  refine bsupr_le (λ K hK, _),
+  rcases is_compact.elim_finite_subcover K.2 _ (λ i, (U i).prop) _ with ⟨t, ht⟩, swap,
+  { convert hK, rw [opens.supr_def, subtype.coe_mk] },
+  rcases K.2.finite_compact_cover t (coe ∘ U) (λ i _, (U _).prop) (by simp only [ht])
+    with ⟨K', h1K', h2K', h3K'⟩,
+  let L : ℕ → compacts G := λ n, ⟨K' n, h1K' n⟩,
+  convert le_trans (h3 t L) _,
+  { ext1, rw [h3K', compacts.finset_sup_val, finset.sup_eq_supr] },
+  refine le_trans (finset.sum_le_sum _) (ennreal.sum_le_tsum t),
+  intros i hi, refine le_trans _ (le_supr _ (L i)),
+  refine le_trans _ (le_supr _ (h2K' i)), refl'
+end
+
+/-- This is "unbundled", because that it required for the API of `induced_outer_measure`. -/
```

Comment:
I think that the docstring should explain the statement of the lemma before it explains why the lemma uses unbundled arguments.